### PR TITLE
Add removed-in tag to `li-reference` and `li-reference-list` metadata plugins

### PR DESCRIPTION
**Pull request by JordiVM**
Relations:
- Related PRs: https://github.com/livingdocsIO/livingdocs-server/pull/6186

### CODE REVIEW

**Review by ajwild — APPROVED** [review]

**Comment by JordiVM** [comment]
I will include the Breaking change message into the Release cut PR, using editor's and server's commit message.

**Comment by JordiVM** [comment]
I wasn't sure how to do it. It made sense to me to just have the breaking change, since the deprecation doesn't imply code change. But if we have done it with both on other places, I can add it here too :)

File: content/reference/document/metadata/plugins/li-reference-list.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ menus:
     parent: Metadata Plugins List
 summary: |
   {{< deprecated-in "release-2023-09" >}}. Use [`li-document-references`]({{< ref "/reference/document/metadata/plugins/li-document-references" >}}) instead.
+  {{< removed-in "release-2023-11" >}}. Use [`li-document-references`]({{< ref "/reference/document/metadata/plugins/li-document-references" >}}) instead.
```
Comment (ajwild) [inline content/reference/document/metadata/plugins/li-reference-list.md:9]: Can you check if this looks ok.
```suggestion
  {{< deprecated-in "release-2023-09" block >}}
  {{< removed-in "release-2023-11" block >}}
  Use [`li-document-references`]({{< ref "/reference/document/metadata/plugins/li-document-references" >}}) instead.
```
I don't think we need to repeat "Use li-document-references instead".

File: content/reference/document/metadata/plugins/li-reference.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ menus:
     parent: Metadata Plugins List
 summary: |
   {{< deprecated-in "release-2023-09" >}}. Use [`li-document-reference`]({{< ref "/reference/document/metadata/plugins/li-document-reference" >}}) instead.
+  {{< removed-in "release-2023-11" >}}. Use [`li-document-reference`]({{< ref "/reference/document/metadata/plugins/li-document-reference" >}}) instead.
```
Comment (ajwild) [inline content/reference/document/metadata/plugins/li-reference.md:9]: Same as above.

**Comment by ajwild** [comment]
Sorry, I didn't really think about where these deprecation and removed messages were being displayed. I think you were right the first time. Lets only have the removed in shortcodes in this summary, otherwise it looks a bit busy. Sorry for causing extra work :grimacing:

**Review by ajwild — APPROVED** [review]

**Pull request merged**